OpenRA.Mods.Common/Traits/ThrowsParticle.cs

```diff
@@ -53,8 +53,8 @@ class ThrowsParticle : ITick
 		int tick = 0;
 		int length;
 
-		float facing;
-		float rotation;
+		int facing;
```

Comment:
Could you make these `WAngle`, and then do `() => facing.Angle / 4` in the Animation ctor?  That makes it clearer what these are for, and has a side-effect of giving you a bit more precision.